tests/test_p2p_daemon.py

```diff
@@ -23,33 +25,131 @@ def is_process_running(pid: int) -> bool:
     return subprocess.check_output(cmd, shell=True).decode('utf-8').strip() == RUNNING
 
 
-@pytest.fixture()
-def mock_p2p_class():
-    P2P.LIBP2P_CMD = "sleep"
-
-
-def test_daemon_killed_on_del(mock_p2p_class):
-    start = perf_counter()
-    p2p_daemon = P2P('10s')
+@pytest.mark.asyncio
+async def test_daemon_killed_on_del():
+    p2p_daemon = await P2P.create()
 
     child_pid = p2p_daemon._child.pid
     assert is_process_running(child_pid)
 
     del p2p_daemon
```

Comment:
nit: `del` [is not guaranteed](https://stackoverflow.com/a/1481512) to call `p2p_daemon.__del__` immediately (though it usually happens), so hypothetically, the test may fail even with the correct implementation.

I suggest to make the `_kill_child` method public and call it here instead of `del p2p_daemon`, or at least explicitly call `p2p_daemon.__del__()`.

Reply:
Done